wls-exporter-core/src/main/java/com/oracle/wls/exporter/AuthenticatedCall.java

```diff
@@ -46,13 +71,57 @@ public WebClient createWebClient() {
         final WebClient webClient = webClientFactory.createClient();
         webClient.addHeader("X-Requested-By", "rest-exporter");
 
-        setAuthentication(webClient);
+        webClient.setAuthentication(context.getAuthenticationHeader());
+        getCookies(context.getAuthenticationHeader()).forEach(c -> webClient.addHeader(COOKIE_HEADER, c));
```

Comment:
Can you please describe the concurrency requirements on this data structure? For instance, could one thread be adding a new cookie value while another thread is iterating to remove expired cookies?

Reply:
Typically the exporter is called about once every five seconds, and the cookie lists are keyed by the credentials. That means that a collision on the cookie list would only happen if two requests with the same credentials happen simultaneously at the time that a cookie is expiring. If a conflict did happen, we might miss sending a cookie, which would mean adding a second with the same name. 

It looks to me though this could easily be avoided simply by synchronizing all cookie operations on the static COOKIES map.

Reply:
Perfect. Thanks.